### PR TITLE
Fix: multiple fixes around classes

### PR DESCRIPTION
**Pull request by hhugo**

### CODE REVIEW

**Comment by hhugo** [comment]
fix #247 247

**Comment by jberdine** [comment]
@hhugo, how can you fix issue #n+5 in pr #n, some kind of magic... :-)

And is this a "Fix"? Tests look like "Improve" or am I overlooking something.

**Comment by hhugo** [comment]
The main purpose of this PR was not to fix 247.

This PR fixes (at least) two issues:
- `class a = object end and class b = object end` was written as `class a = object endand class b = object end` (not space between `end` and `and`)
- doc comments were printed ambiguously. (warning 50)

File: src/Fmt_ast.ml [hunk 1]
```diff
@@ -1896,7 +1896,7 @@ and fmt_class_structure c ~ctx ~parens ?ext self_ fields =
            $ cmts_after_self
            $ fmt_if Poly.(fields <> []) "@;<1000 0>"
            $ hvbox 0
-               (list fields "@\n" (fun cf -> fmt_class_field c ctx cf)) )
+               (list fields "\n@\n" (fun cf -> fmt_class_field c ctx cf)) )
```
Comment (pakoito) [inline src/Fmt_ast.ml]: 🎉

File: src/Fmt_ast.ml [hunk 2]
```diff
@@ -1922,7 +1922,7 @@ and fmt_class_signature c ~ctx ~parens ?ext self_ fields =
            $ cmts_after_self
            $ fmt_if Poly.(fields <> []) "@;<1000 0>"
            $ hvbox 0
-               (list fields "@\n" (fun cf -> fmt_class_type_field c ctx cf))
+               (list fields "\n@\n" (fun cf -> fmt_class_type_field c ctx cf))
```
Comment (pakoito) [inline src/Fmt_ast.ml]: 🎉

**Pull request merged**